### PR TITLE
Problem: state overrides are not supported in eth_call

### LINKED ISSUE #368: Problem: state overrides are not supported
__Proposal:__ support state overrides in multiple json-rpc calls

__Current behavior:__ [What currently happens]

__Desired behavior:__ [What you would like to happen]

__Use case:__ [Why is this important (helps with prioritizing requests)]

Requests may be closed if we're not actively planning to work on them.

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #368

Solution:
- add the feature

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (ante) [#310](https://github.com/crypto-org-chain/ethermint/pull/310) Support blocking list of addresses in mempool.
 * (evm) [#328](https://github.com/crypto-org-chain/ethermint/pull/328) Support precompile interface.
 * (statedb) [#333](https://github.com/crypto-org-chain/ethermint/pull/333) Support native action in statedb, prepare for precompiles.
+* (rpc) [#]() Support state overrides in eth_call.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* (rpc) [#369](https://github.com/crypto-org-chain/ethermint/pull/369) Support state overrides in eth_call.
```

File: tests/integration_tests/test_call.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from web3 import Web3
+from web3._utils.contracts import encode_transaction_data
+from hexbytes import HexBytes
+
+from .utils import CONTRACTS
+
+
+def test_state_override(cronos):
+    state = 100
+    w3: Web3 = cronos.w3
```
Comment (mmsqe) [inline tests/integration_tests/test_call.py]:
```suggestion
def test_state_override(ethermint):
    state = 100
    w3: Web3 = ethermint.w3
```

File: x/evm/statedb/statedb.go [hunk 3]
```diff
@@ -385,6 +385,14 @@ func (s *StateDB) SubBalance(addr common.Address, amount *big.Int) {
 	}
 }
 
+func (s *StateDB) SetBalance(addr common.Address, amount *big.Int) {
+	if err := s.ExecuteNativeAction(common.Address{}, nil, func(ctx sdk.Context) error {
```
Comment (mmsqe) [inline x/evm/statedb/statedb.go:389]: do we need check negative amount here?

Comment (yihuang) [reply x/evm/statedb/statedb.go:389]: I think no.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**